### PR TITLE
Some API changes

### PR DESCRIPTION
**Pull request by thibault-martinez**
https://github.com/iotaledger/iota.go/pull/625

### CODE REVIEW

File: sdk/src/wallet/operations/syncing/outputs.rs [hunk 1]
```diff
@@ -60,7 +60,7 @@ where
                     output_id: output_with_meta.metadata().output_id().to_owned(),
                     metadata: *output_with_meta.metadata(),
                     output: output_with_meta.output().clone(),
-                    is_spent: output_with_meta.metadata().is_spent(),
+                    is_spent: output_with_meta.metadata().spent().is_some(),
```
Comment (Thoralf-M) [inline sdk/src/wallet/operations/syncing/outputs.rs]: How about adding `.is_spent()` which calls `.spent().is_some()` inside?

File: sdk/src/wallet/update.rs [hunk 2]
```diff
@@ -69,7 +69,8 @@ where
                     wallet_data.unspent_outputs.remove(&output_id);
                     // Update spent data fields
                     if let Some(output_data) = wallet_data.outputs.get_mut(&output_id) {
-                        output_data.metadata.set_spent(true);
+                        // TODO wat do?
```
Comment (Thoralf-M) [inline sdk/src/wallet/update.rs]: Guess we just don't update it here

File: sdk/src/wallet/operations/syncing/outputs.rs (hunk 1 above)
Comment (thibault-martinez) [reply sdk/src/wallet/operations/syncing/outputs.rs]: Sure

File: sdk/src/types/api/core.rs [hunk 3]
```diff
@@ -173,6 +174,31 @@ pub struct BaseTokenResponse {
...
+    /// The total stake of the pool, including delegators.
+    #[serde(with = "string")]
+    pool_stake: u64,
+    /// The stake of a validator.
+    #[serde(with = "string")]
+    validator_stake: u64,
+    /// The fixed cost of the validator, which it receives as part of its Mana rewards.
+    #[serde(with = "string")]
+    fixed_cost: u64,
+    /// Shows whether validator was active recently.
```
Comment (Alex6323) [inline sdk/src/types/api/core.rs]:
```suggestion
    /// Shows whether the validator was active recently.
```

File: sdk/src/wallet/operations/syncing/outputs.rs (hunk 1 above)
Comment (thibault-martinez) [reply sdk/src/wallet/operations/syncing/outputs.rs]: Done

File: sdk/src/wallet/update.rs (hunk 2 above)
Comment (thibault-martinez) [reply sdk/src/wallet/update.rs]: Is that not going to cause issues somewhere else? Do we rely on this or it's purely informative ?

Comment (Thoralf-M) [reply sdk/src/wallet/update.rs]: I think we rely on the is_spent value below, but yeah should be double checked to be sure

Comment (Thoralf-M) [reply sdk/src/wallet/update.rs]: Looks like we only use it here https://github.com/iotaledger/iota-sdk/blob/9387a3535dc74978fa1e170f8ac5e9152033a01e/sdk/src/wallet/operations/syncing/transactions.rs#L248 and there I'm not sure if it should be changed or not
Or we have to set some 0x0000 values for the case where the output is spent and pruned so we don't get this information anymore

maybe leave the TODO for now

**Review by Thoralf-M — DISMISSED** [review]

File: sdk/src/wallet/update.rs (hunk 2 above)
Comment (thibault-martinez) [reply sdk/src/wallet/update.rs]: Let me create an issue and update the TODO then

File: sdk/src/types/api/core.rs [hunk 4]
```diff
@@ -173,6 +174,31 @@ pub struct BaseTokenResponse {
...
+    #[serde(with = "string")]
+    validator_stake: u64,
+    /// The fixed cost of the validator, which it receives as part of its Mana rewards.
+    #[serde(with = "string")]
+    fixed_cost: u64,
+    /// Shows whether the validator was active recently.
+    active: bool,
+    /// The latest protocol version the validator supported.
+    latest_supported_protocol_version: u8,
+    // The protocol hash of the latest supported protocol of the validator.
```
Comment (Alex6323) [inline sdk/src/types/api/core.rs]:
```suggestion
    /// The protocol hash of the latest supported protocol of the validator.
```

**Review by Thoralf-M — APPROVED** [review]

File: sdk/src/types/api/core.rs [hunk 5]
```diff
@@ -173,6 +174,31 @@ pub struct BaseTokenResponse {
     pub decimals: u32,
 }
 
+/// Information of a validator.
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ValidatorResponse {
```
Comment (Alex6323) [inline sdk/src/types/api/core.rs:180]: `ValidatorInfoResponse` would be a kinda better name, similar to `NodeInfo` but :shrug:

Comment (Alex6323) [reply sdk/src/types/api/core.rs:180]: not important

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**